grblas/operator.py

```diff
@@ -186,6 +189,18 @@ def __call__(self, val):
         )
 
 
+class TypedBuiltinIndexUnaryOp(TypedOpBase):
+    __slots__ = ()
+    opclass = "IndexUnaryOp"
+
+    def __call__(self, val, thunk=None):
+        if isinstance(val, InfixExprBase):
```

Comment:
Can we drop this check?  For example, `gb.select.tril(A @ B)` should be fine, right?  We use `output_type(val)` instead of `type(val)` to handle expressions and objects alike.